src/main/kotlin/com/cosmotech/api/containerregistry/ContainerRegistryService.kt

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.containerregistry
+
+import com.cosmotech.api.config.CsmPlatformProperties
+import com.cosmotech.api.exceptions.CsmResourceNotFoundException
+import java.util.Base64
+import org.apache.http.client.methods.HttpUriRequest
+import org.apache.http.client.methods.RequestBuilder
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import org.springframework.http.HttpHeaders
+import org.springframework.stereotype.Service
+
+@Service("csmContainerRegistry")
+class ContainerRegistryService(private val csmPlatformProperties: CsmPlatformProperties) {
+  fun getEndpoint() = csmPlatformProperties.containerRegistry.registryUrl
+
+  fun getCredentials(user: String, password: String) =
+      Base64.getEncoder().encodeToString("$user:$password".toByteArray())
+
+  fun getHeaderAuthorization() =
+      "Basic " +
+          getCredentials(
+              csmPlatformProperties.containerRegistry.registryUserName!!,
+              csmPlatformProperties.containerRegistry.registryPassword!!)
+
+  private fun buildHttpRequest(repository: String): HttpUriRequest {
+    var reqBuilder = RequestBuilder.get()
+    reqBuilder = reqBuilder.setUri(getImageRegistryUri(repository))
+    reqBuilder = reqBuilder.addHeader(HttpHeaders.AUTHORIZATION, getHeaderAuthorization())
+
+    return reqBuilder.build()
+  }
+
+  fun checkSolutionImage(repository: String, tag: String) {
+    val tags = getRepositoryTagList(repository)
+    if (tags.contains("errors") || !doesTheTagExist(tags, tag)) {
+      throw CsmResourceNotFoundException(tags)
+    }
+  }
+
+  fun doesTheTagExist(tags: String, tag: String) = tags.contains(tag)
+
+  private fun getImageRegistryUri(repository: String): String {
+    return csmPlatformProperties.containerRegistry.registryUrl + "/v2/" + repository + "/tags/list"
+  }
+
+  private fun execRequest(repository: String): String {
+    val httpResponse = getHttpClient().execute(buildHttpRequest(repository))
+    return EntityUtils.toString(httpResponse.getEntity())
```

Comment:
I'm not entirely sure how the HTTP request work here, but does this mean we treat the response as a single string text instead of a structured JSON? In particular, is the `contains` call to check for the tag presence made on the entire string response or only the `tags` JSON array entry?

What happens if the value of the tag we check is `name`, `tags` or included in the docker image/repository name? Will we get a wrongly successful check even if no tags with those names exists?